driver/controller.go

```diff
@@ -158,12 +159,19 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 	}
 
 	contentSource := req.GetVolumeContentSource()
-	if contentSource != nil {
-		snapshot := contentSource.GetSnapshot()
-		if snapshot != nil {
-			ll.WithField("snapshot_id", snapshot.GetSnapshotId()).Info("using snapshot as volume source")
-			volumeReq.SnapshotID = snapshot.GetSnapshotId()
+	if contentSource != nil && contentSource.GetSnapshot() != nil {
+		snapshotID := contentSource.GetSnapshot().GetSnapshotId()
```

Comment:
Should we validate that the snapshot ID is not empty? Not sure whether the API call below will return a `NotFound` or some other status when passed an empty ID.

Reply:
Good poing @adamwg. I think it'll be already validated but I'll make sure to test that as well.

Reply:
Fixed with: https://github.com/digitalocean/csi-digitalocean/pull/144/commits/46155ff61267529f745e7e41b9f9e838167f87f1